src/library_memfs.js

```diff
@@ -341,7 +341,7 @@ mergeInto(LibraryManager.library, {
         var contents = stream.node.contents;
         // Only make a new copy when MAP_PRIVATE is specified.
         if ( !(flags & {{{ cDefine('MAP_PRIVATE') }}}) &&
-              (contents.buffer === buffer || contents.buffer === buffer.buffer) ) {
+              (contents.buffer === buffer) ) {
```

Comment:
When reading lines like this, I've been wondering if we should avoid referring to that global variable `buffer` anywhere in runtime code, since it is an awfully generic name. Would `(contents.buffer === HEAPU8.buffer)` read here better?